url7d636e8121_laforge49_JActor2_tgz-pJ8-org_agilewiki_jactor2_core_revisited_EHJ8/src/org/agilewiki/jactor2/core/impl/mtMessages/SyncRequestMtImplWithData.java

```diff
@@ -0,0 +1,123 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.agilewiki.jactor2.core.impl.mtMessages;
+
+import org.agilewiki.jactor2.core.messages.SyncOperation;
+import org.agilewiki.jactor2.core.messages.impl.RequestImplWithData;
+import org.agilewiki.jactor2.core.reactors.Reactor;
+
+/**
+ * Internal implementation of a SyncRequest, with user payload.
+ *
+ * @param <RESPONSE_TYPE> The response value type.
+ *
+ * @author monster
+ */
+public class SyncRequestMtImplWithData<RESPONSE_TYPE> extends
+        SyncRequestMtImpl<RESPONSE_TYPE> implements
+        RequestImplWithData<RESPONSE_TYPE> {
+
+    /** Double parameter 0. */
+    private double doubleParam0;
+
+    /** Double parameter 1. */
+    private double doubleParam1;
+
+    /** Double parameter 2. */
+    private double doubleParam2;
+
+    /** Object parameter 0. */
+    private Object objectParam0;
+
+    /** Object parameter 1. */
+    private Object objectParam1;
+
+    /** Object parameter 2. */
+    private Object objectParam2;
+
+    /**
+     * @param _syncOperation
+     * @param _targetReactor
+     */
+    public SyncRequestMtImplWithData(
+            final SyncOperation<RESPONSE_TYPE> _syncOperation,
+            final Reactor _targetReactor) {
+        super(_syncOperation, _targetReactor);
+    }
+
+    /**
+     * @param _targetReactor
+     */
+    public SyncRequestMtImplWithData(final Reactor _targetReactor) {
+        super(_targetReactor);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public double getDouble(final int index) {
+        if (index == 0) {
+            return doubleParam0;
+        }
+        if (index == 1) {
+            return doubleParam1;
+        }
+        // We assume the use of invalid indexes would have already failed when
+        // calling the setter.
+        return doubleParam2;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void setDouble(final int index, final double newValue) {
+        if (index == 0) {
+            doubleParam0 = newValue;
+        } else if (index == 1) {
+            doubleParam1 = newValue;
+        } else if (index == 2) {
+            doubleParam2 = newValue;
+        } else {
+            throw new IndexOutOfBoundsException(String.valueOf(index));
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public Object getObject(final int index) {
+        if (index == 0) {
+            return objectParam0;
+        }
+        if (index == 1) {
+            return objectParam1;
+        }
+        // We assume the use of invalid indexes would have already failed when
+        // calling the setter.
+        return objectParam2;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void setObject(final int index, final Object newValue) {
+        if (index == 0) {
+            objectParam0 = newValue;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem:**
 This line of code lacks validation when processing input data through the following parameter: &#39;newValue&#39; (index: 1 | type: Object). The parameter is exposed to external callers, because its enclosing class and method are publicly accessible. This means that upstream validation, if it exists, can be bypassed. Other validated parameters: &#39;index&#39;. The same parameter type is validated here for example: url7d636e8121_laforge49_JActor2_tgz-pJ8-org_agilewiki_jactor2_core_revisited_EHJ8/src/org/agilewiki/jactor2/core/impl/mtMessages/AsyncRequestMtImpl.java:385. Malicious, malformed, or unbounded inputs can cause unexpected runtime behavior or crashes, and can slow performance.

**Fix:**
 Add checks to ensure the validity of the parameter&#39;s value, such as testing it for nullness (for example, using the @NonNull annotation as [described in the Lombok library](https://projectlombok.org/features/NonNull)), emptiness, or equality. Or to prevent direct calls to it, [reduce the method&#39;s visibility](https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html).

 Learn more about potential threats and guidance from the [Common Weakness Enumeration website](https://cwe.mitre.org/data/definitions/693.html) and the [OWASP Cheat Sheet series](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html).